service/service.go

```diff
@@ -25,126 +26,131 @@ import (
+	vaultClient   VaultClient
+	router        *mux.Router
+	server        *server.Server
+	errChan       chan error
+	shutdown      time.Duration
+	healthCheck   *healthcheck.HealthCheck
 }
 
 // DatasetClient is an interface to represent methods called to action on the dataset api
 type DatasetClient interface {
 	GetVersion(ctx context.Context, userAuthToken, serviceAuthToken, downloadServiceAuthToken, collectionID, datasetID, edition, version string) (m dataset.Version, err error)
-	healthcheck.Client
 }
 
 // FilterClient is an interface to represent methods called to action on the filter api
 type FilterClient interface {
 	GetOutput(ctx context.Context, userAuthToken, serviceAuthToken, downloadServiceAuthToken, collectionID, filterOutputID string) (filter.Model, error)
-	healthcheck.Client
 }
 
 // VaultClient is an interface to represent methods called to action upon vault
 type VaultClient interface {
 	ReadKey(path, key string) (string, error)
-	healthcheck.Client
 }
 
 // Create should be called to create a new instance of the download service, with routes correctly initialised
-func Create(bindAddr, vaultPath, bucketName, serviceAuthToken, downloadServiceToken, zebedeeURL string,
+func Create(
+	cfg config.Config,
 	dc DatasetClient,
 	fc FilterClient,
 	s3sess *session.Session,
 	vc VaultClient,
-	shutdown, healthCheckInterval, healthCheckRecovery time.Duration,
-	isPublishing bool) Download {
+	buildTime, gitCommit, version string) Download {
 
+	ctx := context.Background()
 	router := mux.NewRouter()
 
 	d := handlers.Download{
 		DatasetClient: dc,
 		VaultClient:   vc,
 		FilterClient:  fc,
 		S3Client:      s3crypto.New(s3sess, &s3crypto.Config{HasUserDefinedPSK: true}),
-		BucketName:    bucketName,
-		VaultPath:     vaultPath,
-		IsPublishing:  isPublishing,
+		BucketName:    cfg.BucketName,
+		VaultPath:     cfg.VaultPath,
+		IsPublishing:  cfg.IsPublishing,
 	}
 
-	router.Path("/downloads/datasets/{datasetID}/editions/{edition}/versions/{version}.csv").HandlerFunc(d.Do("csv", serviceAuthToken, downloadServiceToken))
-	router.Path("/downloads/datasets/{datasetID}/editions/{edition}/versions/{version}.csv-metadata.json").HandlerFunc(d.Do("csvw", serviceAuthToken, downloadServiceToken))
-	router.Path("/downloads/datasets/{datasetID}/editions/{edition}/versions/{version}.xlsx").HandlerFunc(d.Do("xls", serviceAuthToken, downloadServiceToken))
-	router.Path("/downloads/filter-outputs/{filterOutputID}.csv").HandlerFunc(d.Do("csv", serviceAuthToken, downloadServiceToken))
-	router.Path("/downloads/filter-outputs/{filterOutputID}.xlsx").HandlerFunc(d.Do("xls", serviceAuthToken, downloadServiceToken))
+	// Create healthcheck object with versionInfo
+	versionInfo, err := healthcheck.NewVersionInfo(buildTime, gitCommit, version)
+	if err != nil {
+		log.Event(ctx, "initialising healthcheck without versionInfo", log.Error(err))
+	}
+	hc := healthcheck.New(versionInfo, cfg.HealthCheckCriticalTimeout, cfg.HealthCheckInterval)
```

Comment:
Healthcheck should be initialised in main.go, this way you can handle configuration errors on startup and exit the application if there was an issue creating version object for heath check.